opencti-platform/opencti-graphql/src/database/engine.ts

```diff
@@ -2517,62 +2552,29 @@ export const buildLocalMustFilter = (validFilter) => {
   throw UnsupportedError('Invalid filter configuration', validFilter);
 };
 
-const buildSubQueryForFilterGroup = (context, user, inputFilters) => {
-  const { mode = 'and', filters = [], filterGroups = [] } = inputFilters;
-  const localMustFilters = [];
-  // Handle filterGroups
-  for (let index = 0; index < filterGroups.length; index += 1) {
-    const group = filterGroups[index];
-    if (isFilterGroupNotEmpty(group)) {
-      const subQuery = buildSubQueryForFilterGroup(context, user, group);
-      if (subQuery) { // can be null
-        localMustFilters.push(subQuery);
-      }
-    }
-  }
-  // Handle filters
-  for (let index = 0; index < filters.length; index += 1) {
-    const filter = filters[index];
-    const isValidFilter = filter?.values || filter?.nested?.length > 0;
-    if (isValidFilter) {
-      const localMustFilter = buildLocalMustFilter(filter);
-      localMustFilters.push(localMustFilter);
-    }
-  }
-  if (localMustFilters.length > 0) {
-    return {
-      bool: {
-        should: localMustFilters,
-        minimum_should_match: mode === 'or' ? 1 : localMustFilters.length,
-      }
-    };
-  }
-  return null;
-};
-
 // If filter key = entity_type, we should also handle parent_types
 // Example: filter = {mode: 'or', operator: 'eq', key: ['entity_type'], values: ['Report', 'Stix-Cyber-Observable']}
 // we check parent_types because otherwise we would never match Stix-Cyber-Observable which is an abstract parent type
-const adaptFilterToEntityTypeFilterKey = (filter) => {
-  const { key, mode = 'or', operator = 'eq' } = filter;
+const adaptFilterToEntityTypeFilterKey = (filter: any) => {
+  const { key, mode = FilterMode.Or, operator = FilterOperator.Eq } = filter;
   const arrayKeys = Array.isArray(key) ? key : [key];
   if (arrayKeys.length > 1) {
     throw UnsupportedError('A filter with these multiple keys is not supported', { keys: arrayKeys });
   }
   // at this point arrayKeys === ['entity_type']
 
   // we'll build these new filters or filterGroup, depending on the situation
-  let newFilter;
-  let newFilterGroup;
+  let newFilter: Filter | undefined;
+  let newFilterGroup: FilterGroup = emptyFilterGroup;
```

Comment:
```suggestion
  let newFilterGroup: FilterGroup | undefined;
```
avoid adding a useless empty filter group in the array filters.filterGroups....